static/js/util/util.js

```diff
@@ -280,3 +292,36 @@ export function makeStrippedHtml(textOrElement) {
     return striptags(textOrElement);
   }
 }
+
+/**
+ * Validate a month number
+ * @param {String} string The input string
+ * @returns {Number|undefined} The valid month or undefined if not valid
+ */
+export function validateMonth(string) {
+  let month = filterPositiveInt(string);
+  if (month === undefined) {
+    return undefined;
+  }
+  if (month < 1 || month > 12) {
+    return undefined;
+  }
+  return month;
+}
+
+/**
+ * Validate a year string is an integer and fits into YYYY
+ * @param {String} string The input string
+ * @returns {Number|undefined} The valid year or undefined if not valid
+ */
+export function validateYear(string) {
+  let year = filterPositiveInt(string);
```

Comment:
cool, that makes sense to check for.

what about this:

```js
let numberFilter = string => (
  Number(string.replace(/\D/,""))
);
```

a little more compact, I guess? haha. your call.